src/main/java/randoop/generation/ForwardGenerator.java

```diff
@@ -275,7 +412,22 @@ private ExecutableSequence createNewUniqueSequence() {
     }
 
     // Select a StatementInfo
-    TypedOperation operation = Randomness.randomMember(this.operations);
+    TypedOperation operation;
+
+    // If bloodhound is enabled, choose the next operation while considering the methods' weights.
+    if (GenInputsAbstract.enable_blood_hound) {
+      operation = Randomness.randomMemberWeighted(this.operationSimpleList, methodWeights);
+    } else {
+      operation = Randomness.randomMember(this.operations);
+    }
+
+    // Update the number of times this method was selected for a new sequence.
+    incrementInMap(methodSelections, operation);
```

Comment:
If this is only used in bloodhound, should it also be in the "if" statement?  Or, abstract this all into a method that updates values and returns a random member.